src/core/client/admin/components/UserHistoryDrawer/UserDrawerAccountHistory.tsx

```diff
@@ -153,10 +179,12 @@ const UserDrawerAccountHistory: FunctionComponent<Props> = ({ user }) => {
                 {formatter.format(history.date)}
               </TableCell>
               <TableCell className={styles.action}>
-                {history.kind === "suspension" ? (
+                {history.kind === "suspension" && (
                   <SuspensionAction {...history.action} />
-                ) : (
-                  <BanAction {...history.action} />
+                )}
+                {history.kind === "ban" && <BanAction {...history.action} />}
+                {history.kind === "username" && (
+                  <UsernameChangeAction {...history.action} />
```

Comment:
Maybe you should create a function that lives outside this one that has a `switch (history.kind)` to keep things clean?